modules/packages/OrderedMap.chpl

```diff
@@ -0,0 +1,755 @@
+    }
+  }
+
+  pragma "no doc"
+  proc _checkValType(type valType) {
+    if isGenericType(valType) {
+      compilerWarning("creating a orderedMap with value type " +
+                      valType:string);
+      if isClassType(valType) && !isGenericType(borrowed valType) {
+        compilerWarning("which now means class type with generic management");
+      }
+      compilerError("orderedMap value type cannot currently be generic");
+    }
+  }
+
+
+  /*
+    The shared nilable class helps us to find one element with the key
+    and without specifying the value
+    See `contains`
+  */
+  pragma "no doc"
+  class _valueWrapper {
+    var val;
+  }
+  /*
+    This doesn't indicate any comparing.
+    Just to make (keyType, shared _valueWrapper?) comparable.
+  */
+  proc <(a: shared _valueWrapper?, b: shared _valueWrapper?) {
+    return false;
+  }
+  proc >(a: shared _valueWrapper?, b: shared _valueWrapper?) {
+    return false;
+  }
+
+  record orderedMap {
+    /* Type of orderedMap keys. */
+    type keyType;
+    /* Type of orderedMap values. */
+    type valType;
+
+    /* If `true`, this orderedMap will perform parallel safe operations. */
+    param parSafe = false;
+
+    /* The comparator used to compare keys */
+    var comparator: record = defaultComparator;
+
+    // TODO: Maybe we want something like record optional for this?
+    pragma "no doc"
+    type _eltType = (keyType, shared _valueWrapper?);
+
+    /* The underlying implementation */
+    pragma "no doc"
+    var _set: orderedSet(_eltType, parSafe);
+
+
+    //TODO: Maybe we should use the lock from the underlying implementation
+    pragma "no doc"
+    var _lock$ = if parSafe then new _LockWrapper() else none;
+
+    pragma "no doc"
+    inline proc _enter() {
+      if parSafe then
+        _lock$.lock();
+    }
+
+    pragma "no doc"
+    inline proc _leave() {
+      if parSafe then
+        _lock$.unlock();
+    }
+
+    pragma "no doc"
+    record _keyComparator {
+      var comparator: record;
+      proc compare(a, b) {
+        return comparator.compare(a[0], b[0]);
+      }
+    }
+
+    /*
+      Initializes an empty orderedMap containing keys and values of given types.
+
+      :arg keyType: The type of the keys of this orderedMap.
+      :arg valType: The type of the values of this orderedMap.
+      :arg parSafe: If `true`, this orderedMap will use parallel safe operations.
+      :type parSafe: bool
+      :arg comparator: The comparator used to compare keys.
+    */
+    proc init(type keyType, type valType, param parSafe = false,
+              comparator: record = defaultComparator) {
+      _checkKeyType(keyType);
+      _checkValType(valType);
+
+      this.keyType = keyType;
+      this.valType = valType;
+      this.parSafe = parSafe;
+      this.comparator = comparator;
+      this._eltType = (keyType, shared _valueWrapper(valType)?);
+
+      this._set = new orderedSet(_eltType, false, new _keyComparator(comparator)); 
```

Comment:
Today on master, this line is problematic: `_set` is declared as `orderedSet(_eltType, parSafe)` in the `orderedMap` definition. But here, you are assigning a `orderedSet(_eltType, false)` to it. This doesn't compile because of type mismatch. It may have been compiling before, but I think the compiler's behavior today is more correct.

The solution is to leave the `_set` s declaration a generic `orderedSet` and let the compiler figure out the right type for it.

I'm planning to do that change post-merge.